Documentation/planning/arcade-powered-source-build/intermediate-nupkg.md

```diff
@@ -0,0 +1,36 @@
+# The source-build intermediate NuGet package
+
+To transfer source-built prebuilts from one repo to the next, we need a
+transport format. NuGet packages are a proven method to do so, including
+built-in caching, secure restore and push supported by Arcade, and versioning.
+
+## The monster package
+
+We can't publish individual source-built NuGet packages (nupkgs) to the feed
+directly. They would conflict with Microsoft-built packages with the same IDs
+and versions. These packages also have a reasonable risk of accidentally being
+used by devs not familiar with source-build, poisoning the NuGet caches on their
+systems.
+
+To address both of these, source-build should assemble a monster nupkg that
+contains all the nupkgs and other artifacts produced by the build.
+
+* `source-build.dotnet-runtime.linux-x64.5.0.0-alpha.1.12345.nupkg/`
+  * `Microsoft.NETCore.App.Ref.5.0.0-alpha.1.12345.nupkg`
+  * `Microsoft.NETCore.App.Runtime.linux-x64.5.0.0-alpha.1.12345.nupkg`
+  * `dotnet-runtime-5.0.0-alpha.1.12345-linux-x64.tar.gz`
+  * ...
+
+Official builds publish the monster nupkg, and downstream repos restore the
```

Comment:
I think more is needed here on 2 things.  
1. When are "official" intermediate packages built and pushed to the feed?  I'm assuming this happens on the rolling build for the repo.  Also, when this fails, how is it recovered and how does it impact downstream builds, i.e. if the package doesn't exist, does the downstream repo build use prebuilts or fail on a missing dependency? 

2. Versioning for these?  I see you specified the version for contained nupkgs as the version of the intermediate package.  That should maybe be specified.

Reply:
> 1. When are "official" intermediate packages built and pushed to the feed? I'm assuming this happens on the rolling build for the repo.

The official build is a specific type of build--depending on the repo it may be per commit, on a schedule, etc.

> Also, when this fails, how is it recovered and how does it impact downstream builds, i.e. if the package doesn't exist, does the downstream repo build use prebuilts or fail on a missing dependency?

When it fails, the official build fails--this way we never need to handle it being missing. I would say we shouldn't try to handle it being missing because that just means people are free to leave their source-build job broken without caring.

> 2. Versioning for these? I see you specified the version for contained nupkgs as the version of the intermediate package. That should maybe be specified.

Good idea. Yep, it just follows the standard Arcade version for the repository. (It shouldn't ever stabilize, since it's a transport package--this helps some with respinning.)